ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/CommandDeduplicationIT.scala

```diff
@@ -3,35 +3,987 @@
+import java.{time, util}
+
+import com.daml.error.ErrorCode
+import com.daml.error.definitions.LedgerApiErrors
+import com.daml.grpc.GrpcStatus
+import com.daml.ledger.api.SubmissionIdGenerator
+import com.daml.ledger.api.testtool.infrastructure.Allocation._
+import com.daml.ledger.api.testtool.infrastructure.Assertions._
+import com.daml.ledger.api.testtool.infrastructure.FutureAssertions._
+import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
+import com.daml.ledger.api.testtool.infrastructure.ProtobufConverters._
+import com.daml.ledger.api.testtool.infrastructure.participant.{
+  CompletionResponse,
+  Features,
+  ParticipantTestContext,
+}
+import com.daml.ledger.api.testtool.suites.CommandDeduplicationIT.{
   DelayMechanism,
+  StaticTimeDelayMechanism,
   TimeDelayMechanism,
 }
-import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantTestContext
+import com.daml.ledger.api.v1.admin.config_management_service.TimeModel
+import com.daml.ledger.api.v1.command_service.SubmitAndWaitRequest
+import com.daml.ledger.api.v1.command_submission_service.SubmitRequest
+import com.daml.ledger.api.v1.commands.Commands.DeduplicationPeriod
+import com.daml.ledger.api.v1.completion.Completion
+import com.daml.ledger.api.v1.completion.Completion.{
+  DeduplicationPeriod => CompletionDeduplicationPeriod
+}
+import com.daml.ledger.api.v1.ledger_offset.LedgerOffset
+import com.daml.ledger.api.v1.version_service.DeduplicationPeriodSupport.OffsetSupport
+import com.daml.ledger.client.binding.Primitive.Party
+import com.daml.ledger.test.model.DA.Types.Tuple2
+import com.daml.ledger.test.model.Test.{Dummy, DummyWithAnnotation, TextKey, TextKeyOperations}
+import com.daml.lf.data.Ref
+import com.daml.lf.data.Ref.{LedgerString, SubmissionId}
+import com.daml.logging.LoggingContext
+import com.daml.timer.Delayed
+import io.grpc.Status.Code
+import org.slf4j.{Logger, LoggerFactory}
 
-import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+import scala.util.{Failure, Success}
 
-/** Command deduplication tests for participant side deduplication
-  * Should be disabled for ledgers that have committer side deduplication enabled (KV)
-  */
 final class CommandDeduplicationIT(
     timeoutScaleFactor: Double,
-    ledgerTimeInterval: FiniteDuration,
     staticTime: Boolean,
-) extends CommandDeduplicationBase(
-      timeoutScaleFactor,
-      ledgerTimeInterval,
-      staticTime,
-    ) {
-
-  override def runWithDeduplicationDelay(
-      participants: Seq[ParticipantTestContext]
+) extends LedgerTestSuite {
+
+  private[this] val logger: Logger = LoggerFactory.getLogger(getClass.getName)
+  private implicit val loggingContext: LoggingContext = LoggingContext.ForTesting
+  val deduplicationDuration: FiniteDuration = scaledDuration(2.seconds)
```

Comment:
Nit: Does it need to be public? If so, I would move it to the companion object and use the constant naming convention.

Reply:
You are correct, this was kept from the merge. Will make it private. We can't move it because it has to be scaled and that needs the param from the constructor.